### PR TITLE
Transition to the new Images

### PR DESCRIPTION
**Pull request by timholy**
Here's the official pull request for https://github.com/timholy/Images.jl/issues/542.

### CODE REVIEW

File: src/ImageMagick.jl [hunk 1]
```diff
@@ -115,39 +118,22 @@ function load_(file::Union{AbstractString,IO,Vector{UInt8}}; ImageType=Image, ex
         end
     end
     # Allocate the buffer and get the pixel data
-    buf = Array(T, sz...)
+    buf = Array{T}(sz)
```
Comment (musm) [inline src/ImageMagick.jl:121]: `Vector{T}(sz)` probably would be better

File: src/libmagickwand.jl [hunk 2]
```diff
@@ -281,7 +297,7 @@ function getimageproperties(wand::MagickWand,patt::AbstractString)
         error("Pattern not in property names")
     else
         nP = convert(Int, numbProp[1])
-        ret = Array(Compat.ASCIIString, nP)
+        ret = Array{String}(nP)
```
Comment (musm) [inline src/libmagickwand.jl]: same as above

File: src/libmagickwand.jl [hunk 3]
```diff
@@ -372,7 +388,7 @@ relinquishmemory(p) = ccall((:MagickRelinquishMemory, libwand), Ptr{UInt8}, (Ptr
 function queryoptions(pattern::AbstractString)
     nops = Cint[0]
     pops = ccall((:MagickQueryConfigureOptions, libwand), Ptr{Ptr{UInt8}}, (Ptr{UInt8}, Ptr{Cint}), pattern, nops)
-    ret = Array(Compat.ASCIIString, nops[1])
+    ret = Array{String}(nops[1])
```
Comment (musm) [inline src/libmagickwand.jl]: same as above

File: src/ImageMagick.jl (hunk 1 above)
Comment (timholy) [reply src/ImageMagick.jl:121]: Images are usually two-dimensional :smile:

Comment (musm) [reply src/ImageMagick.jl:121]: oops, my mistake

Comment (musm) [reply src/ImageMagick.jl:121]: Is `buf` always 2d?

Comment (timholy) [reply src/ImageMagick.jl:121]: No, I just meant they are multidimensional, not one-dimensional. Some TIFF files, for example, have a 3rd dimension (and that's properly handled here).

File: test/constructed_images.jl [hunk 4]
```diff
@@ -87,150 +88,152 @@ facts("IO") do
...
-        c = reinterpret(BGRA{UFixed8}, [0xf0884422]'')
+    @testset "Alpha" begin
+        c = reinterpret(BGRA{N0f8}, [0xf0884422]'')
         fn = joinpath(workdir, "alpha.png")
     	ImageMagick.save(fn, c)
         C = ImageMagick.load(fn)
-        if !ontravis
-            # disabled on Travis because it has a weird, old copy of
+        if !ontravis || !Base.is_linux()
+            # disabled on Linux Travis because it has a weird copy of
```
Comment (tkelman) [inline test/constructed_images.jl:106]: have you tried `dist: trusty` ?

Comment (timholy) [reply test/constructed_images.jl:106]: Didn't know you could do that. Will try sometime, see #63.

**Pull request merged**